Packages/io.chainsafe.web3-unity.web3auth/Runtime/WalletGUI/Scripts/Web3AuthWalletGUIUIManager.cs

```diff
@@ -184,8 +185,9 @@ private IEnumerator CheckHoldPrivateKeyButtonInput()
     /// </summary>
     private void SetPrivateKey()
     {
-        var web3AuthWallet = (Web3AuthWallet)Web3Accessor.Web3.ServiceProvider.GetService(typeof(Web3AuthWallet));
-        privateKeyText.text = web3AuthWallet.Key;
+        var accountProvider = (AccountProvider)Web3Accessor.Web3.ServiceProvider.GetService(typeof(AccountProvider));
```

Comment:
Does this mean that now you'd be _easily_ be able to acess the private key from all of the providers (MM, WC and Web3Auth) trough the account provider ? Obv not in this PR but later on?

Reply:
If we later/in the future add a provider that uses in process like web3auth we can simply add it as a wallet provider

Reply:
this is an implementation I inherited from the [Web3Auth GUI](https://github.com/ChainSafe/web3.unity/pull/1019), you can kind of see that I only changed the service I'm getting and I'm not sure if I can simply make the Monobehaviour injectable I would kind of need to refactor the whole thing which is out of scope but definitely needs attention, we had to let the Web3Auth GUI PR in to meet deadlines.

Reply:
plus `Web3Accessor` is part of the Packages now (has been for while), I even had a todo where I was gonna suggest renaming it to `ChainSafe` singleton object and basically have everything accessible from there xD

Reply:
It is part of the package, but we can't use it for the SDK code for the reasons I described earlier. We can only use it for samples. I don't understand what this class is for, is this a functionality we provide with the SDK?

Reply:
yeah the class is for Web3Auth Wallet GUI, more info [here](https://github.com/ChainSafe/web3.unity/pull/1019).

Reply:
Web3Accessor was intended to be used just for samples but I don't know how to inject dependencies into Monobehaviours without a Unity DI framework. And we can't reference this object before services are built because this object only gets instantiated once web3auth successfully logs in